telemetry/prometheus/metrics.go

```diff
@@ -28,9 +30,9 @@ type metrics struct {
 	cfg *Config
 
 	gaugeVecs     map[string]*prometheus.GaugeVec
-	counterVecs   map[string]*prometheus.CounterVec
+	counterVecs   *rwstore.RWMap[string, *prometheus.CounterVec]
```

Comment:
The other choice is using sync.Map, which may be more suitable for our use case: write (initialized) only once, but will read many times afterwards.

Try to using existing RWMap struct - but if sync.Map is more preferred, happy to update.

Reply:
Checked the doc a bit. RLock is also a cheap ops - let's use it